test_public_network.py

```diff
@@ -112,25 +115,60 @@ def test_public_network_port_security(approach, two_servers_in_same_subnet):
     attacker.assert_run('sudo apt update --allow-releaseinfo-change')
     attacker.assert_run('sudo apt install -y curl ettercap-text-only tcpdump')
 
-    # The attacker starts poisoning the environment
-    mitm = approach['mitm']
-
-    v4 = victim.ip('public', 4)
-    v6 = victim.ip('public', 6)
+    ipv4 = victim.ip('public', 4)
+    ipv6 = victim.ip('public', 6)
+    # Get first IPv6 link local address
+    ll_addr_v6 = ip_interface(victim.output_of(
+        "ip -6 -brief addr show scope link | awk '{ print $3 }' | head -1"
+    )).ip
+    ipv4_gateway = victim.gateway('public', 4)
+    ipv6_gateway = victim.gateway('public', 6)
+    if ip_version == 4:
+        victim_ip = ipv4
+        curl_params='-4'
+        mtm_method = 'arp'
+    elif ip_version == 6:
```

Comment:
I think we can just use `else` here. The chances that we support something larger than IPv6 in this test suite is very slim. The only other place where we check for IPv4 in this code base, we also just `else` to IPv6.

Reply:
I don't care much. I slightly prefer my version because it make it explicit what the else part is about. But I changed the code to follow your suggestion.